File: credit.c
#include <cs50.h>
#include <stdio.h>

void print_brand(long long c, int i);
int main(void)
{
    long long n = get_long_long("Number: ");
    long long c = n;
    int x;
    int i;
    int total1 = 0;
    int total2 = 0;

    for(i=1;n>0;i+=1)
    {
        x = n%10;
        n = n/10;
        if(i%2 == 0)
        {
            x = x*2;
            if(x > 9)
            {
                total1 = total1 + x - 9;
            }
            else
            {
                total1 += x;
            }
        }
        else{
            total2+=x;
        }
    }

    if((total1 + total2)%10 == 0)
    {
        print_brand(c, i);
    }
    else
    {
        printf("INVALID\n");
    }
}

void print_brand(long long c, int i)
{

    while(c>=100)
    {
        c = c/10;
    }

    i = i-1;

    if((c == 34 || c==37) && i == 15)
    {
        printf("AMEX\n");
    }
    else if(c/10 == 4 && (i==13 || i==16))
    {
        printf("VISA\n");
    }
    else if(c>50 && c<56 && i == 16)
    {
        printf("MASTERCARD\n");
    }
    else
    {
        printf("INVALID\n");
    }
}

File: Python/pset6/credit.py
import cs50 as cs

def main():
    n = cs.get_int("Number: ")

    l = [int(x) for x in str(n)]
    l.reverse()

    total1 = 0
    total2 = 0

    for i in range(0,len(l)):

        if i%2 == 0:
            total1+=l[i]

        else:
            j = l[i]*2

            if j > 9:
                total2+=(j - 9)

            else:
                total2+=j

    if (total1 + total2)%10 == 0:
        check(l)

    else:
        print("INVALID")

def check(l):
    l.reverse()
    length = len(l)

    if length == 15 and l[0] == 3 and (l[1] == 4 or l[1] == 7):
        print("AMEX")

    elif length == 16 and l[0] == 5 and (l[1] == 1 or l[1] == 2 or l[1] == 3 or l[1] == 4 or l[1] == 5):
        print("MASTERCARD")

    elif l[0] == 4 and (length == 13 or length == 16):
        print("VISA")

    else:
        print("INVALID")

main()
